recipes/open62541/all/conanfile.py

```diff
@@ -0,0 +1,307 @@
+                self.requires("mbedtls/2.16.3-gpl")
+
+        if self.options.discovery == "With Multicast":
+            self.requires("pro-mdnsd/0.8.4")
+
+    def _patch_sources(self):
+        for patch in self.conan_data["patches"][self.version]:
+            tools.patch(**patch)
+
+    def configure(self):
+        if not self.options.subscription:
+            if self.options.subscription_events:
+                raise ConanInvalidConfiguration(
+                    "Open62541 requires subscription option")
+
+        if not self.options.discovery:
+            if self.options.discovery_semaphore:
+                raise ConanInvalidConfiguration(
+                    "Open62541 discovery sempahore option requires discovery option to be enabled")
+
+        if tools.Version(self.version) <= "1.1.0":
+            if self.options.encryption == "openssl":
+                raise ConanInvalidConfiguration(
+                    "Lower Open62541 versions than 1.1.0 do not support openssl")
+
+            if self.options.multithreading != "None":
+                raise ConanInvalidConfiguration(
+                    "Lower Open62541 versions than 1.1.0 do not fully support multithreading")
+
+            if self.options.web_socket:
+                raise ConanInvalidConfiguration(
+                    "Lower Open62541 versions than 1.1.0 do not fully support websockets")
+
+            if self.options.cpp_compatible:
+                raise ConanInvalidConfiguration(
+                    "Lower Open62541 versions than 1.1.0 are not cpp compatible due to -fpermisive flags")
+
+            if self.settings.compiler == "clang" and tools.Version(self.settings.compiler.version) >= "9":
+                raise ConanInvalidConfiguration(
+                    "Open62541 does not support Clang 9.x.x and later compiler versions")
+
+        if self.settings.compiler == "clang":
+            if tools.Version(self.settings.compiler.version) <= "4":
+                raise ConanInvalidConfiguration(
+                    "Older clang compiler version than 4.0 are not supported")
+
+        if self.options.pub_sub != False and self.settings.os != "Linux":
+            raise ConanInvalidConfiguration(
+                "PubSub over Ethernet is not supported for your OS!")
+
+        if self.options.web_socket:
+            self.options["libwebsockets"].with_ssl = self.options.encryption
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+        if not self.options.cpp_compatible:
+            del self.settings.compiler.cppstd
+            del self.settings.compiler.libcxx
+
+    def source(self):
+        archive_name = self.name + "-" + self.version
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename(archive_name, self._source_subfolder)
+
+        submodule_filename = os.path.join(
+            os.path.dirname(__file__), 'submoduledata.yml')
```

Comment:
You can also use here the [`self.recipe_folder`](https://docs.conan.io/en/latest/reference/conanfile/attributes.html?highlight=recipe_folder#recipe-folder)